src/main.rs

```diff
@@ -63,6 +74,216 @@ async fn init_maker(
     }
 }
 
+// TODO: consider introducing components for this (also balance updates below),
+//  Rough idea: Extend MidMarketRate (or introduce new struct) to give us an async next function, hide the fact that it is using channels behind the next function of the component.
+#[allow(clippy::type_complexity)]
+fn init_rate_updates(
+    update_interval_secs: u64,
+) -> (
+    Pin<Box<dyn Future<Output = comit::Never> + Send>>,
+    Receiver<anyhow::Result<MidMarketRate>>,
+) {
+    let (mut sender, receiver) =
+        futures::channel::mpsc::channel::<anyhow::Result<MidMarketRate>>(5);
+
+    let future = async move {
+        loop {
+            let rate = get_btc_dai_mid_market_rate().await;
+
+            if let Err(e) = sender.try_send(rate) {
+                tracing::trace!("Error when sending rate update from sender to receiver.")
+            }
+
+            Delay::new(Duration::from_secs(update_interval_secs)).await;
+        }
+    };
+
+    (Box::pin(future), receiver)
+}
+
+#[allow(clippy::type_complexity)]
+fn init_bitcoin_balance_updates(
+    update_interval_secs: u64,
+    wallet: bitcoin_wallet::Wallet,
+) -> (
+    Pin<Box<dyn Future<Output = comit::Never> + Send>>,
+    Receiver<anyhow::Result<bitcoin::Amount>>,
+) {
+    let (mut sender, receiver) =
+        futures::channel::mpsc::channel::<anyhow::Result<bitcoin::Amount>>(5);
+
+    let future = async move {
+        loop {
+            let balance = wallet.balance().await;
```

Comment:
:thinking: We could keep track of the last value we sent through the channel and only send again if the new value we get from the wallet is different.

Reply:
This would apply to all of these updates.

Reply:
We can, but that means we store state twice, once in the main loop and once in the Maker.

Reply:
I think we should let the maker decide whether the balance/rate changed and what to do from there.

For example, a rate could cahnge within the spread and the maker could decide not to do anything about it.